platforms/nuttx/src/bootloader/common/bl.c

```diff
@@ -1107,9 +1107,10 @@ bootloader(unsigned timeout)
 			continue;
 		}
 
-		// we got a command worth syncing, so kill the timeout because
-		// we are probably talking to the uploader
-		timeout = 0;
+		// We got a sync command as well as a get_device command, we are very likely talking to the uploader.
```

Comment:
Why not use the  same constructs already in the file

#define STATE_ALLOWS_BOOTLOADER 
`if (bl_state & STATE_ALLOWS_BOOTLOADER ) == STATE_ALLOWS_BOOTLOADER `

Reply:
Dunno, probably cause I wasn't aware of them. Let me try that.

Reply:
@davids5 done and force pushed!